### ISSUE TITLE
Non-working/Buggy DOS games list

### ISSUE THREAD

**Issue by DSN9990** [joncampbell123/dosbox-x#1809]
Games (FIXED):

1. Screamer Rally - hangs with blinking cursor DOS prompt when running setup program, but installs fine.

2. Grand Theft Auto - sound setup fails, does not save configuration, also install hangs when I try to install the game again (after the game is installed).

3. Descent II - CD track 1 stops and does not continue playing anymore.

Games (NOT WORKING/BUGGY):

1. Cannon Fodder (CD) - Gives an SoundBlaster initialization error both in install program when trying to test sound and when trying to run the game (after the intro ends).

Config:

```
[sdl]
fullresolution=desktop
output=direct3d
autolock=false
mouse_emulation=always

[dosbox]
machine=svga_s3
vmemsize=1
memsize=16
vesa modelist cap=16
allow low resolution vesa modes=true
allow explicit 24bpp vesa modes=true
allow high definition vesa modes=true
allow unusual vesa modes=true
allow 32bpp vesa modes=true
allow 24bpp vesa modes=true
allow 16bpp vesa modes=true
allow 15bpp vesa modes=true
allow 8bpp vesa modes=true
allow 4bpp vesa modes=true
allow 4bpp packed vesa modes=true
allow tty vesa modes=true
ignore extended memory bit=true

[cpu]
core=dynamic
cputype=pentium
cycles=max

[dos]
hard drive data rate limit=0
ver=5.0

```

Now using latest (0.83.5 64Bit SDL1).

**Comment by DSN9990** [comment]
Screenshots:

Formula Karts:
![dosinst_001](https://user-images.githubusercontent.com/69558812/90756040-45b88c80-e2b2-11ea-8ac0-a7b5b82acf0f.png)

Screamer Rally:
![command_000](https://user-images.githubusercontent.com/69558812/90756092-5406a880-e2b2-11ea-8983-f2c2da219755.png)

and more to go...

**Comment by DSN9990** [comment]
@joncampbell123 Can you test/reproduce this?

**Comment by joncampbell123** [comment]
@DSN9990 If I can find the game online, I'll try.

However first... did you run LOADFIX before running the game?

**Comment by DSN9990** [comment]
No. I tried now and it did not work also. Same problem.

Here it's the link to the game: [https://www.old-games.ru/game/download/5701.html]

**Comment by Wengier** [comment]
I have checked the game "Formula Karts" from the link you posted. Looks like it is a problem with the downloaded MDF image itself, not a problem with DOSBox-X (BTW: it happens in vanilla DOSBox too). The game installer tries to open the file DOC\README_E.TXT, which cannot be found in the MDF image, and thus the error. If I manually add the DOC\README_E.TXT file to the MDF image, then the installation will succeed. So there is nothing to fix from the DOSBox-X side.

**Comment by DSN9990** [comment]
Can you attach the file README_E.TXT for me here to add to the image?

**Comment by DSN9990** [comment]
You checked Screamer Rally? It's working?

**Comment by Wengier** [comment]
The content of the README_E.TXT file does not matter - just create a text file named  "README_E.TXT" and add it to the DOC folder of the image (as I did), then it should work.

As for Screamer Rally, both the installer and the setup program worked for me. I just need to wait for a few seconds before the graphical interface of the setup program showed up.

**Comment by DSN9990** [comment]
It now showed this:

![command_001](https://user-images.githubusercontent.com/69558812/90815520-357bce00-e301-11ea-973e-21a32b1bc6fe.png)

**Comment by DSN9990** [comment]
For me the Screamer Rally setup program freezes with a blinking cursor when running it.

**Comment by Wengier** [comment]
Formula Karts apparently requires the real game CD, and I don't think the downloaded CD image is a genuine game CD, or required files such as README_E.TXT won't be missing from it.

The Screamer Rally setup program also shows a blinking cursor when running it on my computer in the beginning, but after a few seconds (up to ~10s) the graphical interface will show up. I tried with the default setting too. Please check your configuration if this is not the case.

**Comment by DSN9990** [comment]
It still did not work, I waited more than 10s and it did not work also.
What is your config?

Description: Instead of showing SETUP in the DOSBox-X title bar when running it, it shows COMMAND anymore.

**Comment by Wengier** [comment]
The default config file works for me too:

![image](https://user-images.githubusercontent.com/8216923/90819027-5e9a5f80-e2fd-11ea-8bc6-6c3e123333d1.png)

**Comment by DSN9990** [comment]
It still hangs with the default config showing COMMAND in the title bar:

![hang](https://user-images.githubusercontent.com/69558812/90819952-0bc5a580-e307-11ea-85a3-ca488855fab6.png)

**Comment by Wengier** [comment]
@DSN9990 I think you selected the 3D version during installation. In your case, after you run "SETUP", try to keep typing the "Y" key until DOSBox-X title bar shows "ST". Hope this helps.

**Comment by Wengier** [comment]
The 3DFX version of the game apparently requires DOSBox-X 0.83.5 and the setting "glide=true" in order to work. A 32-bit SDL1 build of the current DOSBox-X 0.83.5 code is available from:

Link: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

You can find 3dfx Voodoo setup guide from the Wiki page:

https://github.com/wengier/dosbox-x-wiki/wiki/Guide%3ASetting-up-3dfx-Voodoo-in-DOSBox%E2%80%90X

**Comment by DSN9990** [comment]
@joncampbell123 Can you give me an alternative link to download Formula Karts?

**Comment by DSN9990** [comment]
@Wengier Can you give me an screenshot of Formula Karts?

**Comment by Wengier** [comment]
I have already fixed the issue of mounting Grand Theft Auto MDF image. The updated Windows build is available from:

Link: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

As for the issue of saving sound config, I believe you are installing the game to a mounted local drive instead of a mounted disk image. And this reveals another bug of FreeDOS's XCOPY: it copies ALL attributes of the source files to the destination including the read-only attribute, and as a result, you cannot write to the DIG.INI file in the mounted local drive because it is read-only. The solution is to either use MS-DOS's XCOPY during installation, or manually remove the read-only attribute of the file C:\GTA\GTADOS\DIG.INI by using the command ```ATTRIB -R C:\GTA\GTADOS\DIG.INI```, or install the game to a mounted disk image instead. Hope this helps.

**Comment by Wengier** [comment]
As for the no sound issue in Desert Strike/Jungle Strike (Issue #1600) you mentioned, @joncampbell123 may want to look at it.

**Comment by Wengier** [comment]
Sound Blaster can be enabled for Desert Strike & Jungle Strike with a simple hack. Try this updated Windows build:

Link: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

**Comment by DSN9990** [comment]
I'm trying to install Jungle Strike CD now and the console triggers a ton of these errors when INSTALL.EXE tries to detect CDROM speed:

1046888 ERROR CPU:Illegal Unhandled Interrupt Called 0

Also DOSBox-X stucks with this screen and stops responding:
![install_000](https://user-images.githubusercontent.com/69558812/91005477-5e27f000-e5ad-11ea-8aef-cda579131ad1.png)

**Comment by rderooy** [comment]
Regarding Jungle Strike, this is a known issue and a work around was posted on Vogons a long time ago: https://www.vogons.org/viewtopic.php?f=33&t=17149&p=272960&hilit=jungle%20strike#p272960

Still, it would be better to fix the issue such that the work around is no longer needed.

**Comment by DSN9990** [comment]
@Wengier Can you fix this since you fixed other issues above?

**Comment by DSN9990** [comment]
I noticed a very slow blinking cursor where DOSBox-X gets stuck, and it hangs after a few seconds.

**Comment by Wengier** [comment]
@DSN9990 The Jungle Strike INSTALL.EXE does work fine for me on DOSBox-X when it tries to detect CDROM speed (no hangs), even though it does indeed hang when I tried it on vanilla DOSBox, DOSBox ECE and dosbox-staging. I downloaded the game ISO from:

https://www.myabandonware.com/game/jungle-strike-235#download

A screenshot for running Jungle Strike INSTALL.EXE on DOSBox-X:

![image](https://user-images.githubusercontent.com/8216923/91062700-07e19e00-e5fb-11ea-8a43-5aceef13bc2f.png)

**Comment by DSN9990** [comment]
@Wengier Wing Commander: Privateer CD tested from Old-Games.RU does not work for me, DOSBox-X "reboots" as soon as I run PRIV.EXE. I used my own config and I am not aware of that issue.

**Comment by Wengier** [comment]
@DSN9990 Yes, it looks like "Wing Commander: Privateer" does not like EMS memory, so you have to set "ems=false" in dosbox-x.conf file before running the game.

**Comment by DSN9990** [comment]
Star Wars TIE Fighter CD (also downloaded from Old-Games.RU) crashes when trying to run it:

![command_002](https://user-images.githubusercontent.com/69558812/91364156-80e81d80-e7d4-11ea-96c5-7b06f5fa9742.png)

I'll try to investigate this issue...

**Comment by Wengier** [comment]
Depending on the method to install and run the said game (I tried minimal installations a few times for faster testing), I did see the crash, but could not reproduce it consistently - it may crash the first time, but then run fine the second and third time, and crashed again the fourth time. Due to lack of time I have not yet concluded a pattern for this.

**Comment by DSN9990** [comment]
@Wengier Tested another game from Old-Games.RU (Inferno: The Odyssey Continues, CD version including CD audio), seems to work fine, but flickers a lot both in the opening scene and in game.

**Comment by DSN9990** [comment]
@Wengier I tested WipEout and it works fine but doesn't have sound, only music (I downloaded an ISO from GamesNostalgia.com). Can you fix this?

**Comment by Wengier** [comment]
@DSN9990 I have tried the WipEout game downloaded from the site you mentioned, and I think I cannot hear sound if the volume of the game music is at the maximum (perhaps because the music is very loud by default). But if I lower the volume of the game music from "Audio options", then I can indeed hear sound, such as the sound when I move among the game options. Maybe this is simply by design, so I am not really sure if there is anything to fix for this.

**Comment by DSN9990** [comment]
Thanks. I was wrong. Can you fix the flickering in "Inferno: The Odyssey Continues" if possible?

**Comment by Wengier** [comment]
I can confirm that flickering occurs in the opening scene of the game "Inferno: The Odyssey Continues” (I have not tried in game), but on my computer at least I can only notice it if I concentrate on the screen. So this issue appears relatively minor.

**Comment by DSN9990** [comment]
I did not find 0.83.5 for Windows on the Releases page. Can you give me an Windows installer for it?

**Comment by joncampbell123** [comment]
@DSN9990 No VS2019 build this month. MinGW builds are chugging along right now.

**Comment by Wengier** [comment]
@DSN9990 The Windows installer should be ready in a day. I will need to compile VS builds myself this time.

**Comment by Wengier** [comment]
@DSN9990 If you do not want to wait, the DOSBox-X 0.83.5 Windows installer (not yet code-signed) is already available below. It includes VS2019 x86/x64/ARM/ARM64 binaries built by myself and MinGW32/64 binaries built by @joncampbell123. I am currently stilling waiting for it to be code-signed so that no SmartScreen warning will be shown when you run the installer.

[DOSBox-X-0.83.5-setup.exe](https://github.com/Wengier/dosbox-x-wiki/raw/master/DOSBox-X-0.83.5-setup.exe)

**Comment by Wengier** [comment]
@DSN9990 The Windows installer is now code-signed. Thanks @emendelson for code-signing it.

**Comment by DSN9990** [comment]
Can you give me a update to 0.83.6 called "dosbox-x-update.zip"?

**Comment by Wengier** [comment]
@DSN9990 Sure! Here it is (built with FluidSynth support):

Link: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

**Comment by DSN9990** [comment]
I found one game that has strange behavior, System Shock CD. It doesn't want to install, no matter what program I run, see below:

![command_003](https://user-images.githubusercontent.com/69558812/92996951-e41cc400-f4e5-11ea-8374-9c7650156d74.png)

Also, if I run INSTALL.EXE on the cd-rom, it says that it requires Microsoft Windows:

![command_004](https://user-images.githubusercontent.com/69558812/92996991-2b0ab980-f4e6-11ea-83fa-055518426153.png)

I am not sure what is the DOS installer of it.

**Comment by rderooy** [comment]
@DSN9990 This is an issue with System Shock, not DOSBox-X. The "Enhanced" CD version simply expects you to run INSTALL.EXE from Windows 95, even though it is a DOS game.

Instead try the following;
```
D:
CD INST
INITIAL.EXE
```
I think that will get you past the problem.

**Comment by DSN9990** [comment]
@Wengier Can you update the file dosbox-x-update.zip to your latest commit? I want to test the CD player you added.

**Comment by Wengier** [comment]
@DSN9990 The new file is here:

Link: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

**Comment by DSN9990** [comment]
@Wengier I tried an simulation game called Darker some days ago and it did not work. It gets stuck with a blinking cursor when I run INSTALL.COM to install the game.

**Comment by Wengier** [comment]
@DSN9990 I just tried the game myself and found that it installs and runs fine with the current code. I downloaded it from:

https://www.myabandonware.com/game/darker-2dn#download

**Comment by DSN9990** [comment]
@Wengier The link you posted have the same ISO I downloaded from Old-Games.RU. What's your config? Do I need an specific setting?

**Comment by Wengier** [comment]
@DSN9990 I have tried with the default setting (-defaultconf) too, and it installs normally. A screenshot:

![image](https://user-images.githubusercontent.com/8216923/94041391-2ec8f680-fd98-11ea-970f-4cecd47e0be3.png)

**Comment by DSN9990** [comment]
@Wengier Can you give me a link to download Gravis Ultrasound drivers?

**Comment by Wengier** [comment]
Gravis UltraSound sound driver for DOS and Windows 3.x:

https://winworldpc.com/product/gravis-ultrasound-drivers/gf1-dos-win31

**Comment by DSN9990** [comment]
Thanks!!! I got full sound in Duke3D. It sounds perfectly.

**Comment by inukaze** [comment]
Hi there @DSN9990 can you make a little favor? can you download and test "The Elder Scrolls : Arena" can be free downloaded from https://elderscrolls.bethesda.net/es/daggerfall/

The thing is i installed like explanations like this -> https://www.instructables.com/id/How-to-Install-Daggerfall-on-DOSBox/

Everything looks like working normally, the thing is when i try to save the game, the screen just goes fully black, can you test if happen to you too?

**Comment by DSN9990** [comment]
@inukaze I was not able to reproduce this issue. I tried it and it works just fine for me when I either save/load a game. I used Gravis Ultrasound in the setup program. I downloaded a floppy version from Old-Games.RU (and a patch to update the game to 1.06).

**Comment by inukaze** [comment]
Well i had a "cd" version and i use "Sound Blaster" xD because i don't know how to configure gravis ultrasound for full working in dosbox xD, the patch on the cd version i had (include a spanish fan translation) is for put the game ni the version 2.13. (but i don't know if really that is an official version can be applied using a patch)

**Comment by DSN9990** [comment]
@Wengier I found that the volume of the GUS is too low compared to the SB16. Is there a way to fix this?

**Comment by rderooy** [comment]
@DSN9990 did you try using the MIXER command?

![command_000](https://user-images.githubusercontent.com/17701098/94304503-eebc6c00-ff6f-11ea-8afd-7a7fe01893d1.png)

**Comment by Wengier** [comment]
@DSN9990 Now I see why when you try to install the Darker game it got stuck with a blinking cursor. It is because you used the "dynamic" core. If you use the normal core it will install just fine.

**Comment by DSN9990** [comment]
@Wengier Cannon Fodder CD doesn't work for me, gives an message "Error initializing Sound Blaster card". Can you fix this issue?

**Comment by DSN9990** [comment]
The GUS had some problems for me, like distorted sound in C&C. Because of this I disabled it and I will use it only for demoscene productions.

**Comment by Wengier** [comment]
@DSN9990 What are the settings you use for GUS?

**Comment by Wengier** [comment]
@DSN9990 Cannon Fodder CD seems to work fine when I select "Sound Blaster" and "Adlib" in the settings before launching the game with CANNON.BAT. I downloaded the game from:

https://wowroms.com/en/roms/dos/cannon-fodder/146532.html

**Comment by DSN9990** [comment]
@Wengier I found that Cannon Fodder CD crashes only when the Dynamic core is selected. If I choose the Normal core the game will run. Can you fix this problem so the game can run in Dynamic core?

**Comment by Wengier** [comment]
@DSN9990 The dynamic core has issues not just for this game, but also for Windows 95/98 in general too, so I think significant work is required for improving this issue. Why not just use the normal core instead, which has the maximum compatibility with DOS games and applications?

**Comment by joncampbell123** [comment]
@DSN9990 Dynamic core cannot run anything that handles page faults in a manner other than handling the fault and returning to the fault address. For that reason, dynamic core is generally fine for DOS extenders like CWSDPMI (and Quake) and Windows 3.1.

However the preemptive multitasking and page fault handling in Windows 95 doesn't work well with dynamic core. Windows 95/98/ME allows task switching during a page fault, which is at odds with DOSBox SVN's general design and the dynamic core's expectations of page fault handling. Furthermore dynamic core does not restart instructions properly in that case. CPU state can get corrupted because half an instruction executes before a fault is processed and returning from the fault can result in half of this state and half the other state of the task it switched from.

There's a reason normal core in DOSBox-X was heavily modified to use a C++ exception (GuestPageFault) instead of recursion and why each instruction is designed to restore initial CPU state upon a page fault. This is what allows DOSBox-X's normal core to run Windows 95/98/ME properly.

**Comment by DSN9990** [comment]
I don't use the normal core because it is slower depending on the application (at least for me?).

**Comment by DSN9990** [comment]
The normal core can be optimized for lower-end systems?

**Comment by joncampbell123** [comment]
@DSN9990 Yes, well, dynamic core's design, inherited from DOSBox SVN, doesn't work well with some corner cases like what I described. I wish it did.

I do recall someone on the Vogons forums mentioning that some fix involving the stack pointer can improve dynamic core's compatibility with Windows 95, but I'm not sure what that fix is yet.

**Comment by DSN9990** [comment]
@joncampbell123 DOSBox-X's normal core can be optimized for lower-end systems, if possible?

**Comment by joncampbell123** [comment]
@DSN9990 Maybe. At this time the dynamic core code is borrowed and updated from DOSBox SVN since it is a popular feature.

**Comment by DSN9990** [comment]
@joncampbell123 One more question, which other game is known to use CWSDPMI instead of Quake?

**Comment by joncampbell123** [comment]
@DSN9990 Not sure, most DOS games I test like to use DOS4GW.EXE, but you can tell by whether or not there is a CWSDPMI.EXE in the same directory.

**Comment by DSN9990** [comment]
Oops. I closed and reopened now.

**Comment by DSN9990** [comment]
I tested Terra Nova: Strike Force Centauri in DOSBox-X and I noticed corrupted colors in the main menu:

![__ff_000](https://user-images.githubusercontent.com/69558812/94373235-6b6b5980-00da-11eb-9651-7f86c9eaffa1.png)

I also noticed that in vanilla DOSBox, this problem does not occur so there is a issue in DOSBox-X.


![tnfdosbox](https://user-images.githubusercontent.com/69558812/94373359-5b07ae80-00db-11eb-88db-a015d7f133d5.jpg)

**Comment by joncampbell123** [comment]
@DSN9990 I think I came across this before... some later 90s games that support VESA BIOS modes will use 8-bit DAC mode if available. In 8-bit DAC mode the VGA palette registers hold a full 8-bit value instead of just a 6-bit value.

DOSBox SVN as far as I know does not emulate that. DOSBox-X does. However implementing it and testing it shows that some games decide whether 8-bit DAC mode is available by trying the call and can misprogram the palette if the return value is not exactly what it expects.

Try:

    enable 8-bit dac=false

**Comment by DSN9990** [comment]
Thanks. The option fixed that.

**Comment by joncampbell123** [comment]
@DSN9990 I have a demo copy of that game that I test with which also uses the 8-bit DAC mode and it works properly here. Perhaps the full game uses it differently?

**Comment by DSN9990** [comment]
@joncampbell123 I don't know.

**Comment by DSN9990** [comment]
The graphics rendering in Zone Raiders looks very, very bad:

![raid_000](https://user-images.githubusercontent.com/69558812/94375100-90b29480-00e7-11eb-9e65-26612e1f8b70.png)

![raid_001](https://user-images.githubusercontent.com/69558812/94375111-9f994700-00e7-11eb-835a-5c6c3f62232d.png)

**Comment by joncampbell123** [comment]
DOSBox SVN, and VS2019 builds of DOSBox-X, use the "double" floating point type for FPU emulation, unless you use dynamic core.

MinGW and Linux builds of DOSBox-X use the "long double" floating point type for FPU emulation which can provide the full 80-bit precision the FPU uses.

Some DOS games and demos have code that is sensitive to whether "double" or "long double" floating point precision is available.

**Comment by DSN9990** [comment]
@joncampbell123 Which host CPU do you recommend for running Win95 with normal core in DOSBox-X?

**Comment by joncampbell123** [comment]
32-bit or 64-bit x86/x64 should be fine, just make sure you give Windows 95 enough hard disk space and memory. You can run it with or without IDE emulation. Windows 95 at least here runs at an acceptable speed given 20000-30000 cycles on relatively recent Intel or AMD processors.

Make sure you use normal core all the way through the install process. Dynamic core can screw up the install process.

If any other faults occur, try disabling APM bios emulation. At one time that was also a source of crashes at least with Windows 98/ME.

**Comment by DSN9990** [comment]
@Wengier I got the following error when trying to install WWF In Your House:

![command_006](https://user-images.githubusercontent.com/69558812/94630017-cba8f980-029a-11eb-81f7-cabc8f28cba8.png)

**Comment by Wengier** [comment]
@DSN9990 Where did you download the game from? I tried to download the game from both of the following places, but there isn't even an INSTALL.EXE program in it. I can find SETUP.EXE and WWFIYH.EXE though, both seem to work fine here.

https://www.myabandonware.com/game/wwf-in-your-house-2dz

http://www.abandonia.com/en/downloadgame/407

**Comment by DSN9990** [comment]
@Wengier From Old-Games.RU. I don't know why it comes with INSTALL.EXE.

**Comment by Wengier** [comment]
@DSN9990 I could not test that now, but the posted error seems to suggest that your %TEMP% environment variable is not correctly set. You could check that, such as changing it to a different location what is writable.

**Comment by DSN9990** [comment]
@Wengier I got it working by changing TEMP=C:\WINDOWS\TEMP to TEMP=C:\.

**Comment by DSN9990** [comment]
@Wengier I got a "Divide error" when I go to install Retribution, a game by Gremlin Interactive. I also tried normal core with no luck.

**Comment by Wengier** [comment]
@DSN9990 I downloaded the game from the following URL, and INSTALL.EXE seems to load quite slowly before the full-screen interface shows up, but it does seem to run fine afterwards.

https://www.myabandonware.com/game/retribution-2vc

**Comment by muhul** [comment]
When I run WinkerMander 3, the following error occurs and the Dosbox reboots.

> LOG:   39056101 ERROR BIOS:INT15:Unknown call ax=1209
> LOG: DOSBox-X has switched to max cycles, because of the setting: cycles=auto.
> If the game runs too fast, try a fixed cycles amount in DOSBox-X's options.
> LOG: VCPI:Could not load TR with 10
> LOG: CPU_Interrupt() interrupted
> LOG: CPU_Interrupt() interrupted
> LOG: CPU_Exception: Exception 14 already in progress, triggering double fault instead
> LOG: Warning: PAGING_NewPageFault() more than one level, now using level 2
> LOG: CPU_Exception: Double fault already in progress == Triple Fault. Resetting CPU.
> LOG: Removing UMB block 0xcc00-0xdfff
> LOG: Rebooting the system

I've tried all the ems options, but they're all in the same situation.

**Comment by rderooy** [comment]
Set core=normal in your config file.

**Comment by muhul** [comment]
Thank you for your quick reply.
However, setting core to normal results the same

> 
> LOG:   48758589 ERROR BIOS:INT15:Unknown call ax=1209
> LOG: DOSBox-X has switched to max cycles, because of the setting: cycles=auto.
> If the game runs too fast, try a fixed cycles amount in DOSBox-X's options.
> LOG: VCPI:Could not load TR with 10
> LOG: CPU_Interrupt() interrupted
> LOG: CPU_Exception: Exception 14 already in progress, triggering double fault instead
> LOG: Warning: PAGING_NewPageFault() more than one level, now using level 2
> LOG: CPU_Exception: Double fault already in progress == Triple Fault. Resetting CPU.
> LOG: Removing UMB block 0xcc00-0xdfff
> LOG: Rebooting the system

**Comment by Wengier** [comment]
@muhul I have tried the demo version, and setting ems=false does seem to work here (either core=normal or core=dynamic). You can also enable/disable EMS memory from the "DOS" menu. Setting "ems=true" on the other hand will indeed cause DOSBox-X to reboot.

**Comment by DSN9990** [comment]
@Wengier Can you give me a 0.83.7 x64 VS2019 SDL1 build?

**Comment by muhul** [comment]
@Wengier Thank you for your answer. It's working right.
But even if I use emm386 on my retro machine, it works well, but it's weird that I have problems with dosbox.

**Comment by Wengier** [comment]
@DSN9990 Both 32-bit and 64-bit VS2019 SDL1 builds are available from:

32-bit: [dosbox-x-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-update.zip?raw=true)

64-bit: [dosbox-x-64-update.zip](https://github.com/Wengier/dosbox-x-wiki/blob/master/dosbox-x-64-update.zip?raw=true)

**Comment by Wengier** [comment]
@muhul I suppose there is something wrong with the interactions of the EMS and CPU code, so you do need to disable EMS at this time. Same for the "Wing Commander: Privateer" game mentioned earlier in this thread. @joncampbell123 may know how to fix this.

**Comment by DSN9990** [comment]
@Wengier I got a BSOD in Win95 with the new dynamic_x86 core. Is there any fix? I tried also with APM bios disabled with no success.

**Comment by Wengier** [comment]
@DSN9990 You meant you can boot into the Win95 desktop with the dynamic_x86 core, but see BSOD afterwards? @joncampbell123 mentioned it has something to do with the S3 driver/emulation, and disabling the S3 driver prevents the BSOD. I hope there is a fix for this too.

**Comment by Wengier** [comment]
@DSN9990 The issue with the graphics rendering in Zone Raiders seems to be specific to 64-bit Visual Studio builds. See Issue #1950.

**Comment by Wengier** [comment]
@DSN9990 Actually, when I tested the game in the latest code (64-bit Visual Studio builds), there seems to be no such issue any more. But I can confirm it did occur in the release version 0.83.6 (64-bit Visual Studio builds). It was probably fixed with other code chagnes.

**Comment by DSN9990** [comment]
@Wengier The BSOD in Win95 is already fixed?

**Comment by Wengier** [comment]
@DSN9990 I did not encounter the BSOD in Win98 myself perhaps because I was not using the driver that may cause the BSOD. But @joncampbell123 did see the BSOD and find that it is related to the S3 driver/emulation, and I think he was trying to find a fix for this.

**Comment by muhul** [comment]
[Tyrian2K](https://en.wikipedia.org/wiki/Tyrian_(video_game)#Tyrian_2000_(3.0))'s sound effect keeps making noise.
Noise that does not occur in dosbox-staging is occurring in dosbox-x.
dosbox-x([link](https://drive.google.com/file/d/1HACrbTbnb-HMox98vS465s7KZRc-Dv32/view?usp=sharing))
dosbox-stagiing([link](https://drive.google.com/file/d/1F9bo_Zdu9xsSgsLZXMyAMxuN6FEvg4Rz/view?usp=sharing))
Below is the setup section of the sblast.
> sbtype                                           = sb16
> sbbase                                           = 220
> irq                                              = -1
> mindma                                           = -1
> irq hack                                         = none
> dma                                              = 1
> hdma                                             = 5
> dsp command aliases                              = true
> pic unmask irq                                   = false
> enable speaker                                   = false
> enable asp                                       = false
> disable filtering                                = true
> dsp write buffer status must return 0x7f or 0xff = false
> pre-set sbpro stereo                             = false
> sbmixer                                          = false
> oplmode                                          = auto
> adlib force timer overflow on detect             = false
> oplemu                                           = nuked
> oplrate                                          = 44100
> oplport                                          = 
> hardwarebase                                     = 220
> force dsp auto-init                              = false
> force goldplay                                   = false
> goldplay                                         = true
> goldplay stereo                                  = true
> dsp require interrupt acknowledge                = auto
> dsp write busy delay                             = -1
> blaster environment variable                     = true
> sample rate limits                               = false
> instant direct dac                               = false
> stereo control with sbpro only                   = false
> dsp busy cycle rate                              = -1
> dsp busy cycle always                            = -1
> dsp busy cycle duty                              = -1
> io port aliasing                                 = true

**Comment by Wengier** [comment]
@muhul I think I heard some noise with the game Tyrian 2000 (downloaded from https://www.abandonwaredos.com/abandonware-game.php?abandonware=Tyrian+2000&gid=1886) when I moved among the menu options, but I believe it can only be noticed if the volume of the sound effect is high (at least moderately higher than the music volume). It seems that it would occur with the default setting for earlier DOSBox-X versions as well, but I think it may be possible to silent such noise with some existing config option since there are apparently quite a few of them.

**Comment by Wengier** [comment]
I even tried to run the game Tyrian 2000 in a real DOS PC and see how the sound works. However either SETUP.EXE or TYRIAN.EXE gives the error "Run-time error 200" in a real DOS system, so I have to patch them. But even after that it appears they do not run with the Sound Blaster Live! DOS driver used for the sound card, so I could not check how it works on a real DOS system.

**Comment by muhul** [comment]
I tested it with my Retoro Machine, but I don't know what the actual sound is because it doesn't recognize the sound card, but I don't think there will be any noise at the end of the effect.
Just in case, I tested it with pcem and 86box, but the noise at the end of the effect did not go in.

**Comment by DSN9990** [comment]
@Wengier The BSOD in Win95 is finally fixed? @joncampbell123 has already fixed the issue with S3 emulation?

**Comment by rderooy** [comment]
@DSN9990 I don't think anyone is currently working on these issues.

**Comment by Wengier** [comment]
@DSN9990 As said earlier I did not encounter the BSOD in Win98 myself probably because I was not using the driver that may cause the BSOD. For now you can certainly use a different video driver on your system too and see how it works.

**Comment by DSN9990** [comment]
I'll close this issue.

**Comment by DSN9990** [comment]
I'll close this issue.

**Comment by inukaze** [comment]
@DSN9990 : You had a guide write by yourself with the specific version of win9x and specific things for use for example  Win98 without the BSOD ? the emulated gpu, the video driver, the sound driver, etc . . . ?

**Comment by DSN9990** [comment]
I want to continue since the point I ended. So I will reopen now.

**Comment by DSN9990** [comment]
@Wengier Give me a updated 64Bit MinGW SDL2 build containing the latest code, such as the fix for #2162.

**Comment by Wengier** [comment]
@DSN9990 You can try the DOSBox-X 64-bit MinGW SDL2 binary containing on the latest code from:

* [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by DSN9990** [comment]
@kcgen The CD-DA issue is still happening in Shadow Warrior for me. Only in Alien Trilogy the CD audio loops correctly, I guess, so it may be something that DOSBox-X doesn't handle yet. Since this issue still remains in Shadow Warrior, it will affect other DOS games, too. Here is the version I used (which contains individual .bin tracks, of course) so you may try yourself:
[https://archive.org/details/ShadowWarriorUSA](url)

**Comment by kcgen** [comment]
@DSN9990 , Thanks for confirming Alien Trilogy is working.  Indeed, I only used that as my test case and regression tested others.  Will take a look at Shadow Warrior next!

**Comment by kcgen** [comment]
> Since this issue still remains in Shadow Warrior, it will also affect other DOS games that use CD music, too. Here is the version I used (which contains individual .bin tracks, of course) so you may try yourself:
> [https://archive.org/details/ShadowWarriorUSA](url)

Hi @DSN9990 ,

I tested the updated DOSBox-X with the linked Shadow Warrior.  I began a new game, which also starts the game playing track 3. Track 3 lassts 183 seconds, after which it fades out, and then then game restarts Track 3.  

I also tested Power Slave (per the original report), and the CD audio loops there too.

Can you re-check these?  Thanks.

**Comment by DSN9990** [comment]
@kcgen Sorry, it seems that I was wrong. I noticed the title (main menu) music track was not looping, but it seems that this is the correct behavior of the game, so I made a mistake anyway. Thanks!

**Comment by DSN9990** [comment]
@Wengier I am having a issue with the Config Tool on my end. I always get this E_Exit error when I click "Save & Restart":

![eexiterror](https://user-images.githubusercontent.com/69558812/115982282-3cce3800-a570-11eb-99d8-e3ad437650c8.PNG)

It is very easy to reproduce, just click on "Save..." and then "Save & Restart" in the Config tool:

![saveandrestart](https://user-images.githubusercontent.com/69558812/115982447-7f444480-a571-11eb-81cf-cc628b5e6c2f.PNG)

I also experienced the same issue in the previous MinGW 64 sdl2 build before this one you sent above.

**Comment by Wengier** [comment]
@DSN9990 Yes, I can reproduce the said issue when using the Direct3D output the MinGW SDL2 build, and it appears that the problem only happens with this output in Windows SDL2 builds. Please check out the updated MinGW SDL2 build:

* [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by DSN9990** [comment]
@Wengier Thanks! The issue goes away with this build.

**Comment by DSN9990** [comment]
@Wengier I found an issue with one game, Jurassic Park. Its installer doesn't seem to detect the C drive even it is already mounted:

![install_000](https://user-images.githubusercontent.com/69558812/117490994-fa303680-af45-11eb-88c2-1808196711a6.png)

If I run JP.BAT on the root, it gives this message:

![command_000](https://user-images.githubusercontent.com/69558812/117491106-22b83080-af46-11eb-83f8-58073e34aa52.png)

I got the game here: [https://archive.org/details/JurassicParkUSA](url)

**Comment by Wengier** [comment]
@DSN9990 It seems that I got this behavior everywhere (including real MS-DOS) when using your linked download and running the installer. Perhaps it has specific requirement(s) in order to continue.

**Comment by DSN9990** [comment]
Thanks. Can @joncampbell123 check this out, if possible?

**Comment by rderooy** [comment]
I have Jurassic Park a quick test and ran into the same issue. The screen looks exactly the same with the same Available and Required space and the same missing drive letter. I tried loadfix and limiting the disk space available on C: with -freesize but it made no difference.

The dosbox-x default log also had no errors. dosbox-staging also had the same issue.

**Comment by rderooy** [comment]
According to ripsaw8080 https://www.vogons.org/viewtopic.php?p=169833#p169833

The installer is simply buggy. It cannot handle 26 drives. He says the same thing happens with MS-DOS on a real PC if you set ``lastdrive=z``. And since dosbox uses the Z:\ drive and has ``lastdrive=z`` you run into the issue.

I tried fixing that with dosbox-x as such:
```
mount -z e
config -set lastdrive=e
```
That moves the Z: drive to E: and changes the lastdrive to E. But I still get the same issue.

**Comment by DSN9990** [comment]
@rderooy So, if the installer is faulty, does it mean it is possible to manually install the game? I think manually copying the game to the hard drive can get the game to work. Also, this issue (at least for me) seems to look more or less similar to another game I mentioned some time ago, Gunmetal, but not really sure. The DOS installer in this game is totally broken, unlike the Win9x installer.

**Comment by DSN9990** [comment]
> @rderooy So, if the installer is faulty, does it mean it is possible to manually install the game? I think manually copying the game to the hard drive can get the game to work. Also, this issue (at least for me) seems to look more or less similar to another game I mentioned some time ago, Gunmetal, but not really sure. The DOS installer in this game is totally broken, unlike the Win9x installer.

Wait, according to the link above, it seems that limiting the number of drives (such as 25) can help with this issue. Is there a setting that limits the available drives in the config file?

**Comment by Wengier** [comment]
@DSN9990 There is a LASTDRIV command to change the DOS last drive (or you can set lastdrive from [config] section of the config file). But as @rderooy already mentioned it does not work in DOSBox-X for the purpose of this game.

**Comment by rderooy** [comment]
You can either use the patch that @ripsaw8080 provided on vogons, or you could run real DOS in dosbox-x (making sure you did not set ``lastdrive=z``), install it, and after installation transfer the data from the HDD image to your folder mounted HDD.

To be honest, based on the description of ripsaw8080 I was expecting it to work by simply limiting the drives in dosbox-x, so either that does not work properly, or there is something more to it.

**Comment by Wengier** [comment]
@rderooy You are right that the last drive handling in DOSBox-X was not correct, so I have tried to fix it.

@DSN9990 Please check out the updated 64-bit MinGW SDL1 build. Hopefully it will work after you move the Z drive with a command like `mount -z e` (and make sure you do **not** set `lastdrive=z` in the config file:

* [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

**Comment by DSN9990** [comment]
@Wengier The link you sent above does not work for me. Can you fix it, please?

**Comment by rderooy** [comment]
I can confirm that the Jurassic Park INSTALL.EXE now works if I move the Z: drive. This worked:

```
mount -z y
config -set lastdrive=y
d:
cd jurassic
install.exe
```

**Comment by Wengier** [comment]
@DSN9990 It is the exactly same link that I sent you earlier for the MinGW SDL1 build, which has been updated to match the latest code. The server may be temporarily inaccessible for some reason, but the link is certainly not broken, so please try download it again.

**Comment by Wengier** [comment]
@DSN9990 How about this then:

* [dosbox-x-mingw64.zip](https://github.com/Wengier/dosbox-x-wiki/raw/master/dosbox-x-mingw64.zip)

**Comment by DSN9990** [comment]
@joncampbell123 I am having some weird glitches in Powerslave. This game flickers a lot in certain circumstances, see below:

![ps_001](https://user-images.githubusercontent.com/69558812/118158197-d8292f00-b3f1-11eb-9563-594798260c24.png)

This happens when I get close to walls, but not only this way. Can you check this out when you have time?

**Comment by joncampbell123** [comment]
I would also run the game on a Pentium or higher system with a reasonably fast PCI VGA card (1997-ish maybe) and check to see if the same artifact occurs. Powerslave if I recall came out in 1995 when the average system might have been too slow to trigger that artifact.

**Comment by DSN9990** [comment]
@joncampbell123 Thanks. This fixed it, but it also slows down the game, making it undesirable to play. So, is there another option that can help with these artifacts?

**Comment by joncampbell123** [comment]
the vmemdelay option is configurable. Do smaller values help without causing artifacts?

**Comment by DSN9990** [comment]
Unfortunately no : )

**Comment by DSN9990** [comment]
I found out that a value of 270 is enough to eliminate the artifacts, but it slows down the frame rate anyway. I will test this later in DOSBox-SVN to see if the same artifacts occur.

**Comment by joncampbell123** [comment]
DOSBox SVN also renders the screen one quarter (1/4th) at a time, not one scanline at a time as DOSBox-X does. That does improve performance but at the cost of covering up artifacts like that. It also prevents some scanline effects like palette changes mid-frame from working.

The only exception is machine=vgaonly, in that case DOSBox SVN will also render at scanline resolution.

So if you want to compare both emulators, set machine=vgaonly and run it in DOSBox SVN.

**Comment by DSN9990** [comment]
@joncampbell123 OK, I gave it a try in DOSBox SVN now, and the same behavior occurs as if in DOSBox-X, however it seems to be a little more worse than in DOSBox-X. So, I'm wondering if the same occurs in other emulators such as PCem and 86Box, but probably not, I guess.


> The only exception is machine=vgaonly, in that case DOSBox SVN will also render at scanline resolution.
> 
> So if you want to compare both emulators, set machine=vgaonly and run it in DOSBox SVN. You may find the artifact occurs there as well.


I tried the default machine=svga_s3, and the same flickering occurs as if in DOSBox-X. I haven't tried machine=vgaonly yet. 
But as you mentioned the same artifacts will occur in DOSBox-X regardless of machine=vgaonly and machine=svga_s3 settings 
as well.

**Comment by DSN9990** [comment]
@joncampbell123 Just to ask, is there a possibility of implementing a hack to address such problem in this game (and others) since there are already various hacks in the config file?

**Comment by DSN9990** [comment]
I think this problem is related to the heavy flickering in the game "Inferno: The Odyssey Continues" I mentioned earlier in this thread too.

**Comment by joncampbell123** [comment]
I found a copy of Powerslave.

Flickering does not occur unless the cycle count is cranked up to about 150,000 to 200,000. I can tell by the flowing blood animation behind the ankh in the status area (your health meter I assume) that the game renders as fast as possible and does not limit itself to vertical refresh, but uses VGA page flipping regardless. The higher the cycle count, the faster the blood animation flows. At 200,000 the animation is running very fast and the game engine is likewise rendering too fast.

At that cycle count you're just about emulating the equivalent of a Pentium II 233MHz which is way faster than the programmers probably intended the game to run at. Going by my own experience the average PC in 1996 was a 100MHz 486 or a 100Mhz Pentium.

I expect that if you run the game on something like a 333MHz Pentium III or faster you will probably see the same artifacts and rendering issues.

**Comment by DSN9990** [comment]
@joncampbell123 Which cycles count can match the average speed of a real Pentium 100Mhz anyway? I think the Pentium 100Mhz option (found in the CPU -> Emulate CPU speed menu) does not match 100% of the average speed of a real P100 system, though.

**Comment by joncampbell123** [comment]
enable pci bus=true and cycles=35000 seems to roughly approximate a 100MHz Pentium to me.

**Comment by Wengier** [comment]
I got this when I select the "Pentium 100Mhz" option and run NSSI:

![image](https://user-images.githubusercontent.com/8216923/118206800-4214f900-b431-11eb-82b5-1b717d1add35.png)

The setting `enable pci bus` is set to true here. There is a `cycle emulation percentage adjust` option which allows for adjustment with the "Emulate CPU speed" menu options, although only between -25% and 25%.

**Comment by DSN9990** [comment]
Unfortunately even with 60000 cycles (that does seem to replicate the average speed of a real 100Mhz Pentium) the game does run (but not much) slow and there is still some flickering when you use the - and + keys on the numpad to reduce\increase the screen size, though.
@joncampbell123 has said above that 35000 cycles does also seem to replicate a real P100 Mhz, but that seems to give the average speed of a 66 Mhz Pentium rather than an 100Mhz Pentium.
Also, my config file is also set to `enable pci bus=true` here, too.

**Comment by Wengier** [comment]
I tried on a different computer (also with `enable pci bus=true`, which seem to be the default) and I got the following:

![image](https://user-images.githubusercontent.com/8216923/118208974-afc32400-b435-11eb-91af-77da9b46ab7c.png)

The data of the Pentium 100Mhz etc were provided by @maximus105 earlier.

**Comment by grapeli** [comment]
For me, under DOSBox-X (also -SVN), there is no this type of flickering or similar artifacts in Powerslave.
I can see quite a lot of them in this recording.
https://www.youtube.com/watch?v=cjgXpQknNEQ

I have tried various settings and cannot experience them.
What influences it. Game version, operating system, graphics drivers?

**Comment by joncampbell123** [comment]
@grapeli I can get *serious* flickering by cranking the CPU cycle count up to about 300,000 or higher, so it may be a combination of vmemdelay and the CPU cycles count aka how fast the DOS machine is. I am using a fixed cycles count though, perhaps "max" or "auto" makes it better or worse. Perhaps dynamic core has an effect as well.

The game appears to render as fast as possible without synchronizing to vertical refresh, which may explain the artifacts.

**Comment by joncampbell123** [comment]
@grapeli As I said farther up, the best way to test whether this is a normal artifact of the game is to run it on real DOS hardware. I suspect the same artifacts would be visible if I ran it in DOS on something fast like a Pentium III or higher.

**Comment by grapeli** [comment]
@joncampbell123 
I am using an outdated and slow processor (12 years old).
Powerslave runs very smoothly.

NSSI benchmark in DOSBox-X shows.
CPU ~213000 Dhrystones/s
FPU ~123000 KWhetsones/s

**Comment by grapeli** [comment]
Short video showing the speed of the animation on the screen of the main menu in [PCExhumed](https://gitlab.com/exhumed/exhumed) and Powerslave (dosbox-x).

https://user-images.githubusercontent.com/452325/118327478-3b4fba00-b4f5-11eb-961b-f577071847ee.mp4

**Comment by RNMB15** [comment]
When I was still going to school, I copied the following learning programs from school computers. I didn't find that on the internet;

These are the programs:
[LERNE MIT COMPUTER.zip](https://github.com/joncampbell123/dosbox-x/files/6481476/LERNE.MIT.COMPUTER.zip)

[BECSHAREWAR.zip](https://github.com/joncampbell123/dosbox-x/files/6481470/BECSHAREWAR.zip)

**Comment by DSN9990** [comment]
I was trying Tex Murphy: Under a Killing Moon a few days ago and I got a very, very strange behavior. The game, for some odd reason, simply seems to think I'm running Windows 3.x even I don't have it installed or running. See below:


https://user-images.githubusercontent.com/69558812/118578213-22980c00-b762-11eb-96a7-5bb6549df055.mp4


How this is possible? Perhaps this game is buggy? @joncampbell123 may want to look at this once he has time.

**Comment by joncampbell123** [comment]
Can you bring up the debugger, add a breakpoint for INT 2Fh AH=16h, then run the game?

BPINT 2F 16

Proper Windows detection uses AX=1600h.

http://www.ctyme.com/intr/rb-4495.htm

**Comment by DSN9990** [comment]
@joncampbell123 I am not sure on how to do this. How can I bring up the debugger?

**Comment by joncampbell123** [comment]
@DSN9990 The default mapper shortcut is Alt+Pause, same as DOSBox SVN.

**Comment by Wengier** [comment]
@DSN9990 Where did you download the game? I tried the game demo from below and it seems to run:

https://archive.org/details/Killing_Moon_demo

**Comment by DSN9990** [comment]
@Wengier Here: https://archive.org/details/TexMurphyUnderAKillingMoonUSA

**Comment by DSN9990** [comment]
@Wengier I am using the MinGW64 SDL2 build you provided in #2509. But it appears that this build doesn't include debugging anyway, or if it does, I am not sure where it is.

**Comment by Wengier** [comment]
@DSN9990 Thanks for the link. And you are right that MinGW builds do not currently support the debugger (Visual Studio builds do).

**Comment by Wengier** [comment]
@DSN9990 Please check out the updated Windows builds, which fixed the Windows detection and should hopefully let your game run just fine:

* MinGW SDL1 build: [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)
* MinGW SDL2 build: [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by Wengier** [comment]
@DSN9990 If you want to try the Visual Studio builds which have the debugger, please check out the following 32-bit Visual Studio builds. The "Debug" menu is now a top-level menu right next to the "Help" menu:

* Visual Studio SDL1 build: [dosbox-x-update.zip](http://cndos.fam.cx/dosware/dosbox-x-update.zip)
* Visual Studio SDL2 build: [dosbox-x-sdl2-update.zip](http://cndos.fam.cx/dosware/dosbox-x-sdl2-update.zip)

**Comment by DSN9990** [comment]
@Wengier I can confirm the game working now. Thanks!

**Comment by krixkrax** [comment]
hello, may i report a buggy game too here?
i have problem with "inferno - the odyssey continues".
the game runs very laggy on dosbox-x. its like, its wanna run fast, then something holds it back, uneven stuttering in every second, easy to hear in the opening music.
im using mingw sdl2 on winxp host. (only the sdl2 version working on xp.)
the game runs fine on other dosboxes (0.74, ECE), except on dosbox-x. 
what can be the problem and what settings should i try?
thanks,
t.

**Comment by Wengier** [comment]
@krixkrax Can you provide a URL where you downloaded the game? Perhaps the issue is also related to the build you use (MinGW SDL2), since there are several builds available (even some Visual Studio builds may be unofficially patched to run on WinXP).

**Comment by krixkrax** [comment]
@Wengier hi, i downloaded the game from here: https://www.old-games.com/download/4312/inferno
(cd version - easy setup.)

**Comment by Wengier** [comment]
@krixkrax Downloading from the old-games site is not free. I did find a copy of the game downloaded from www.old-games.ru earlier, and running the game appears to be fine (and smooth) here, even inside Windows XP VM. I tried with the default DOSBox-X configuration with IRQ=5 as usually done (`dosbox-x -defaultconf -set irq=5`).

**Comment by grapeli** [comment]
I was running and it ran smoothly.
The configuration recommended for her is suddenly.
```
[cpu]
cycles=15000
```
It's really not much.

**Comment by krixkrax** [comment]
@grapeli i use to play with 26800 or above.

**Comment by krixkrax** [comment]
@Wengier its a no go for me, i wonder what can be the problem... thanks anyway.

**Comment by DSN9990** [comment]
@krixkrax and @Wengier I tried it again, and reducing the cycles values does help (though slowing down the game a little) with the flickering in both the opening and in game. Thanks!

**Comment by krixkrax** [comment]
@Wengier by the way, the problems about "inferno" i mentioned above: 
i tried it with 08314 sdl2 and all my problems are gone which were exist with 08313.
dunno what caused that uneven speed/stuttering bug in 08313 though.

**Comment by DSN9990** [comment]
I just tried running INSTALL.EXE in DOSBox-X 0.83.13 and it works fine there but in the latest version (0.83.14) it does not. So I guess some recent commit/change is causing some issue with this game anyway. Here is a screenshot (0.83.13):

![install_001](https://user-images.githubusercontent.com/69558812/120868645-d70b9d80-c56a-11eb-8f95-528b7a2effd9.png)

With latest version (0.83.14) it writes the following errors to the log (reporting a triple fault, which causes DOSBox-X to reset):

```
LOG: CPU_Exception: Exception 6 already in progress, triggering double fault instead
LOG: CPU_Exception: Double fault already in progress == Triple Fault. Resetting CPU.
LOG: Warning: C++ exception method is not compatible with dynamic core when emulating reset
LOG: Removing UMB block 0xcc00-0xdfff
LOG: Rebooting the system
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: menuScale=1
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: menuScale=1
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: menuScale=1
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
LOG: Release
LOG: ISA Plug & Play BIOS enabled
LOG: VGA ROM BIOS init callback
LOG: pixratio 1.000, dw false, dh false
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: menuScale=1
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
LOG: DOS keyboard layout loaded with main language code BR for layout br
LOG: XMS: 50 handles allocated for use by the DOS environment
LOG: EMS page frame at 0xe000-0xefff
LOG: COMMAND.COM env size:             720 bytes
LOG: COMMAND.COM environment block:    0x0701 sz=0x002d
LOG: COMMAND.COM main body (PSP):      0x072f sz=0x009a
LOG: COMMAND.COM stack:                0x0749
LOG: pixratio 1.200, dw false, dh false
LOG: Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
LOG: menuScale=1
LOG: Screen report: Method 'Win98base' (1280.000 x 1024.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (
-1.000 x -1.000 DPI)
```

**Comment by Wengier** [comment]
@DSN9990 Which build are you using for the game, e.g. Visual Studio or MinGW, SDL1 or SDL2? Have you tried other builds too?

**Comment by DSN9990** [comment]
@Wengier I'm using the MinGW64 SDL2 build, just like before 0.83.13. I think some change(s) is probably causing a problem anyway.

**Comment by Wengier** [comment]
@DSN9990 You can try set either `cputype=386` or `allow lmsw to exit protected mode=false` (both in [cpu] section) and run the game again.

**Comment by DSN9990** [comment]
@Wengier Thanks. I am now able to run the game just fine now.

**Comment by Wengier** [comment]
The config option "allow lmsw to exit protected mode" is a recent addition to solve another issue (#2549). @joncampbell123 may want to take a look at it.

**Comment by krixkrax** [comment]
hello!
i'm playing "star crusader" (oldgames.ru) nn a 4:3 monitor and i can't set the proper full screen ratio in the game with dosbox-x.
my question is:
is it possible to set the screen like on this screenshot:
![dosbox074](https://user-images.githubusercontent.com/81972013/122511135-14f6d000-d007-11eb-9408-d57ca5b66fd6.JPG)
taken from dosbox074 / ece,
instead of this ratio:
![dosbox-x](https://user-images.githubusercontent.com/81972013/122511290-496a8c00-d007-11eb-9ebe-1c4563635990.JPG)
taken from dosbox-x, which is too stretched vertically?
im using this settings:
output=openglnb
aspect=false.
thanks for any help.

**Comment by Wengier** [comment]
@krixkrax Normal OpenGL modes use the "stretch" mode by default, but the OpenGL pixel-perfect mode does not, and there was previously a discussion to make a setting for this. Have you tried `output=openglpp` (by @ant-222) and see how it works?

**Comment by krixkrax** [comment]
@Wengier 
thanks, openglpp works! although the screen shrinked a little, but the ratio is good. 

![openglpp](https://user-images.githubusercontent.com/81972013/122545301-fb1cb380-d02d-11eb-831e-ae67ffb97d7e.JPG)

**Comment by DSN9990** [comment]
@Wengier and @joncampbell123 I have now made a DOS 6.22 setup successfully, and almost everything appears to be working properly (one such example not working is the low DMA channel not being detected such as reported in #1589), however the only issue I have is the mouse not working properly on the DOS virtual machine. Here is a screenshot from Mouse Manager from Microsoft Mouse 9.01 (as you can see, the cursor seems to be locked at the bottom):

![guest os_001](https://user-images.githubusercontent.com/69558812/124413117-65b83980-dd26-11eb-9695-91ee6e4080fa.png)

Even capturing with Ctrl+F10 it doesn't work. Not sure anyway. Regarding Microsoft Mouse, I'm running with auxdevice=none and serial1=serialmouse settings as well. The drivers I used can be found at: https://winworldpc.com/product/microsoft-mouse/9x

**Comment by DSN9990** [comment]
@joncampbell123 I found out that one game I have, Star Wars: TIE Fighter CD-ROM, simply doesn't work anymore for some reason, it immediately crashes with the same error screen I posted earlier in this thread. But the weird thing is that I remember it worked fine before with 0.83.13/14 (about 2 months ago). I have tried 0.83.14 too and same problem, it crashes anyway. Do I need a specific option/config to allow the game to run just like as before? I used this version: https://archive.org/details/StarWarsTIEFighterCollectorsCDROMUSA
Can you check this out when possible?

**Comment by Wengier** [comment]
@DSN9990 I think you mentioned the crash with the game earlier, which I could reproduce only sometimes. Is there any version that works for the game consistently?

**Comment by joncampbell123** [comment]
Try turning off Gravis Ultrasound emulation. The install program wants to use the Gravis Ultrasound by default, which will not work unless you provide the Gravis drivers and software in C:\ULTRASND. If it does not detect the GUS then it will want to use the Sound Blaster.

**Comment by joncampbell123** [comment]
I got it to crash ONCE on startup, but I can't seem to recreate it. However the music seems to play pretty fast in both Sound Blaster and Gravis Ultrasound configurations. It seems unusually CPU speed sensitive for a 1995 game.

**Comment by DSN9990** [comment]
@Wengier I tried now with -defaultconf switch and it crashes the same way, and as I mentioned above it does not work anymore even with 0.83.14 for some reason, since this is the version I used before to run the game. So I'm thinking there may be a bug in the game itself, which causes it to crash sometimes (or not). @joncampbell123 As you mentioned you got it to crash only once on the first time, but for me this is not the case. Even starting the game for the 2nd/3rd time the game still crashes anyway (at least for me?), so the game seems to be simply buggy in any case.

**Comment by DSN9990** [comment]
Regarding the GUS, it is completely disabled for my config, as I rarely use it, so it may be useful mainly for demoscene, as stated in the wiki guide there are lots of demoscene productions that only support the GUS though.

**Comment by DSN9990** [comment]
@joncampbell123 and @Wengier Sorry, I was completely wrong this time. It seems to crash only if I select "PLAY" from the main setup program after installing the game, otherwise if I choose to exit the installer and manually go to the C:\TIECD to run the game, it works very well. So this is simply a bug in the setup program that "corrupts" the entire game, thus preventing it from running. Thanks!

**Comment by DSN9990** [comment]
@Wengier I am still experiencing random crashes in the game in the latest version, sometimes it works fine, but not always. Not sure why.....

**Comment by DSN9990** [comment]
@Wengier I always get this error when starting Carmageddon 3DFX on low-level Voodoo mode:

![command_001](https://user-images.githubusercontent.com/69558812/126071477-9d2ed6a9-c59a-4ff5-8b81-f97b032220b5.png)

Is there a way to fix this? Which DLL is needed to run the game? GTA 3DFX also has this same error when starting in LLE mode, but, for some reason, Tomb Raider 3DFX works fine even without a external DLL file on low-level mode.

**Comment by Wengier** [comment]
@DSN9990 For low-level Voodoo mode, you will need to use the original 3dfx GLIDE2X.OVL instead of the one provided by DOSBox-X, or else the high-level 3dfx emulation will be assumed and you will need GLIDE2X.DLL in the OS. Hope this helps.

**Comment by DSN9990** [comment]
@Wengier Ok, it seems to work fine now but the only major issue I have is that the game runs too slowly in this mode, unlike Tomb Raider 3Dfx. Can the Voodoo hardware emulation be further optimized?

**Comment by DSN9990** [comment]
I used the 3Dfx glide2x.ovl driver included with the GTA CD in the /3DFX directory as well.

**Comment by Wengier** [comment]
My suggestion is to try the OpenGL mode for the Voodoo hardware emulation rather than the software emulation. Or  use the high-level Glide mode. Hope this helps.

**Comment by DSN9990** [comment]
I used OpenGL mode but it still runs slower compared to Tomb Raider. Is the OpenGL mode "really" faster than software?

**Comment by DSN9990** [comment]
Some time ago I've tried HLE mode as well but I noticed many graphic glitches in Carmageddon 3DFX anyway, but Tomb Raider/GTA 3Dfx worked fine for me in this mode too.

**Comment by Wengier** [comment]
@DSN9990 It is supposed to be faster than software, but there may be exceptions for certain games (I have not confirm this myself). But the high-level Glide mode is definitely the fastest.

**Comment by DSN9990** [comment]
@Wengier As I said above, Carmageddon 3DFX runs very, very poorly for me in Glide mode. There is a lot of annoying flickering in the movies/menu screens and in game, fonts are totally missing:

https://user-images.githubusercontent.com/69558812/126077676-c6ee2bc3-25da-4fc1-844f-376729da4dc8.mp4

https://user-images.githubusercontent.com/69558812/126077693-ba1bd300-9f32-4e8d-beed-262fcc172264.mp4

Is it supposed to be this way???

**Comment by Wengier** [comment]
@DSN9990 What Glide wrapper are you using? Perhaps try a different one?

**Comment by DSN9990** [comment]
As you can see, there's also no ingame music in this mode, since it plays properly on the software rendering version.

**Comment by DSN9990** [comment]
Oh forgot. I used DgVoodoo x64 with MinGW64 DOSBox-X. Maybe this will help? I'm not 100% sure though.

**Comment by DSN9990** [comment]
@Wengier One more question, which is the recommended host CPU for running software low-level Voodoo emulation? The wiki page says that it needs a very poweful CPU clock to run properly.

**Comment by Wengier** [comment]
@DSN9990 Maybe you can try nGlide with 32-but DOSBox-X build and see how it works? For the question of host CPU, perhaps @rderooy can answer for this.

**Comment by DSN9990** [comment]
@Wengier I have also tried nGlide with 32Bit MinGW SDL2 DOSBox-X in the last hours and I got some different bugs/glitches which I will post a short video soon. @rderooy Can you explain me the minimum CPU requirements needed to run software LLE 3dfx emulation when you have time? Despite being very CPU-demanding, it seems to be more accurate than OpenGL or Glide modes, too.

**Comment by rderooy** [comment]
I cannot be clearer on it. Just that a higher clock frequency will help more than core count as the code does not really take advantage of more cores.

**Comment by DSN9990** [comment]
@Wengier I want to try the latest changes in DOSBox-X, so give me a updated MinGW64 SDL2 build. Thanks!

**Comment by Wengier** [comment]
@DSN9990 Please try the updated MinGW 64-bit SDL1 build for now:

* [dosbox-x-mingw64.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64.zip)

Hope this helps.

**Comment by grapeli** [comment]
@DSN9990
For me, Carmageddon under DOSBox-X with `voodoo_card=opengl` (or `glide`) reaches 20fps. Much less when viewed from the cockpit. It is completely unplayable without 3dfx. With the glide under SDL2, the same symptoms you described occur.

It works great under Wine. A short video.

https://user-images.githubusercontent.com/452325/127766863-d1c132a0-1655-4037-90a4-f992bd59d845.mp4

**Comment by krixkrax** [comment]
hello, recently i'm messing with mechwarrior 2 and the game runs fine on dosbox-x.
i decided to make small cue/bin/ogg package of the game.
surprisingly the game doesn't recognize the mounted image.

![mw2](https://user-images.githubusercontent.com/81972013/128624376-584cd35b-14fb-41b5-bc8d-7051b80cab9b.JPG)

i made such images many times before, they all work fine. so no problem with the cue sheet or the mounting.
it's more like dosbox missing something from the raw bin image, i dunno...
anyone experienced such thing before?

**Comment by DSN9990** [comment]
@Wengier As I said above, it is stuck on fullscreen mode forever (at least on Carmageddon) and there are some missing elements on the HUD. Remember, I have recorded this video with nGlide 2.10. See below:


https://user-images.githubusercontent.com/69558812/130534608-855e0d43-90de-4a4a-8d0c-4ca0639b1de7.mp4

**Comment by grapeli** [comment]
This is what Carmageddon 3dfx looks like under dosbox-x 0.83.26-git. Using internal 3dfx emulation (not glide).

https://user-images.githubusercontent.com/452325/168495252-c0427945-28e2-4284-99b7-99dfdb52c9ad.mp4

There is a lot of progress. First of all, the additional graphic elements do not disappear, the second important change is the ability to scale the window. In the latter case, I undone the 1624f04 commit that made it impossible. Scaling additional graphic elements is far from perfect (in my case it is not an integer multiple).
Due to screen capture for video, the average fps drop is about 6-7 frames. In fact, the average is higher, around 21fps. My laptop is very weak.
Core i5-450M 2.4GHz (turbo 2.6GHz) with GPU Intel Gen5 Architecture (Ironlake) 
**On normal hardware, this should run smoothly enough.**

**Comment by grapeli** [comment]
[Dethrace](https://github.com/dethrace-labs/dethrace) is an open clone of Carmageddon. Not everything works yet, but it's worth running and testing.

Under wine, the original works better for me.
